test/functional/p2p_compactblocks_blocksonly.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test that a node in blocksonly mode does not request compact blocks."""
+
+from test_framework.messages import (
+    MSG_BLOCK,
+    MSG_CMPCT_BLOCK,
+    MSG_WITNESS_FLAG,
+    CBlock,
+    CBlockHeader,
+    CInv,
+    from_hex,
+    msg_block,
+    msg_getdata,
+    msg_headers,
+    msg_sendcmpct,
+)
+from test_framework.p2p import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+
+class P2PCompactBlocksBlocksOnly(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 4
+        self.extra_args = [["-blocksonly"], [], [], []]
+
+    def setup_network(self):
+        self.setup_nodes()
+        self.sync_all()
+
+    def build_block_on_tip(self):
+        blockhash = self.nodes[2].generate(1)[0]
+        block_hex = self.nodes[2].getblock(blockhash=blockhash, verbosity=0)
+        block = from_hex(CBlock(), block_hex)
+        block.rehash()
+        return block
+
+    def run_test(self):
+        # Nodes will only request hb compact blocks mode when they're out of IBD
+        for i in range(3):
+            assert not self.nodes[i].getblockchaininfo()['initialblockdownload']
```

Comment:
Should be 4 now that there are 4 nodes? Or maybe even better:

```suggestion
        for node in self.nodes:
            assert not node.getblockchaininfo()['initialblockdownload']
```